src/client.rs

```diff
@@ -119,6 +119,15 @@ where
             #[cfg(not(feature = "embedded-tls"))]
             Err(Error::InvalidUrl(nourl::Error::UnsupportedScheme))
         } else {
+            #[cfg(feature = "embedded-tls")]
+            match self.tls.as_mut() {
+                Some(tls) => Ok(HttpConnection::PlainBuffered(BufferedWrite::new(
+                    buffered_io_adapter::ConnErrorAdapter(conn),
+                    tls.write_buffer,
+                ))),
+                None => Ok(HttpConnection::Plain(conn)),
+            }
+            #[cfg(not(feature = "embedded-tls"))]
             Ok(HttpConnection::Plain(conn))
```

Comment:
Brilliant idea!

Reply:
I'd call it an unfortunate hack but thanks :D